### PR TITLE
Public setter for MetricReaderOptions.PeriodicExportingMetricReaderOptions

### PR DESCRIPTION
**Pull request by alanwest**
https://github.com/open-telemetry/opentelemetry-dotnet/issues/2620#issuecomment-1099384706

### CODE REVIEW

**Comment by alanwest** [comment]
Might have a flakey Prometheus test

```
[xUnit.net 00:00:00.94]     OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterHttpServerTests.PrometheusExporterHttpServerIntegration_NoMetrics [FAIL]
[xUnit.net 00:00:00.95]       System.Net.Http.HttpRequestException : Connection refused
[xUnit.net 00:00:00.95]       ---- System.Net.Sockets.SocketException : Connection refused
[xUnit.net 00:00:00.95]       Stack Trace:
[xUnit.net 00:00:00.95]            at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
[xUnit.net 00:00:00.95]            at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
[xUnit.net 00:00:00.95]            at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
[xUnit.net 00:00:00.95]            at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
[xUnit.net 00:00:00.95]            at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
[xUnit.net 00:00:00.95]            at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
[xUnit.net 00:00:00.95]            at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
[xUnit.net 00:00:00.95]         /home/runner/work/opentelemetry-dotnet/opentelemetry-dotnet/test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterHttpServerTests.cs(159,0): at OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterHttpServerTests.RunPrometheusExporterHttpServerIntegrationTest(Boolean skipMetrics)
[xUnit.net 00:00:00.95]         /home/runner/work/opentelemetry-dotnet/opentelemetry-dotnet/test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterHttpServerTests.cs(42,0): at OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterHttpServerTests.PrometheusExporterHttpServerIntegration_NoMetrics()
[xUnit.net 00:00:00.95]         --- End of stack trace from previous location where exception was thrown ---
[xUnit.net 00:00:00.95]         ----- Inner Stack Trace -----
[xUnit.net 00:00:00.95]            at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
  Failed OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterHttpServerTests.PrometheusExporterHttpServerIntegration_NoMetrics [25 ms]
```

**Review by cijothomas — APPROVED** [review]
a changelog is important as this is going to be stable release in a day!

File: src/OpenTelemetry/Metrics/MetricReaderOptions.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public class MetricReaderOptions
     public MetricReaderTemporalityPreference TemporalityPreference { get; set; } = MetricReaderTemporalityPreference.Cumulative;
 
     /// <summary>
-    /// Gets the <see cref="PeriodicExportingMetricReaderOptions" /> options.
+    /// Gets or sets the <see cref="PeriodicExportingMetricReaderOptions" />.
     /// </summary>
-    public PeriodicExportingMetricReaderOptions PeriodicExportingMetricReaderOptions { get; private set; } = new PeriodicExportingMetricReaderOptions();
+    public PeriodicExportingMetricReaderOptions PeriodicExportingMetricReaderOptions { get; set; } = new PeriodicExportingMetricReaderOptions();
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Metrics/MetricReaderOptions.cs:32]: @alanwest This probably introduces the chance of a null reference exception we need to guard against where we access the value. Alternatively, don't use an auto-prop here so you can do the guard in the setter if it is being set to `null`.

Probably a rare thing we can do in a follow-up.

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry/Metrics/MetricReaderOptions.cs (hunk 1 above)
Comment (alanwest) [reply src/OpenTelemetry/Metrics/MetricReaderOptions.cs:32]: 🤦 yep, good catch. Pushed a fix, could add a test to in a sec.

Comment (alanwest) [reply src/OpenTelemetry/Metrics/MetricReaderOptions.cs:32]: > Alternatively, don't use an auto-prop here so you can do the guard in the setter if it is being set to null.

Maybe this is a good pattern to consider across all our options classes that have nested settings. Could do this in the future.

**Pull request merged**